src/components/SideBar.component.js

```diff
@@ -63,12 +63,13 @@ class SideBar extends BindableHTMLElement {
       if (!this.mapCanvas.options.layers || !this.mapCanvas.options.layers[i] || !this.mapCanvas.jsonResults) {
         continue;
       }
+      const legendText = this.mapCanvas.options.layers[i].name || `Layer ${i + 1}`;
       sidebarLayerContent += `<div class='toggle container' ${ !this.mapCanvas.options.layers[i].legend && "style='display: none;'" }>
         <label class="switch">
-          <input type="checkbox" ${ this.mapCanvas.options.layers[i].visible && "checked"} id="sidebar-layer-${i}">
+          <input role="checkbox" aria-labelledby="${legendText}"  type="checkbox" ${ this.mapCanvas.options.layers[i].visible && "checked"} id="sidebar-layer-${i}">
```

Comment:
I believe `aria-labelledby` should be a reference to an element ID rather than a copy of the label text. I think this should either be `aria-label` (which may or may not be applicable to `<input type="checkbox">`) or it should correspond to the label id. If this isn't specifically used by E2E tests, I would definitely remove this, because it's likely to significantly change this markup in a way that will probably also trigger changes to the related CSS.

Reply:
I removed the `aria-labelledby` and `role` attributes; a better alternative for detecting the controls rendered here was already in place by narrowing of locators and the attributes here were a hold over from trying a different targeted implementation.